app/models/schedule_rule.rb

```diff
@@ -130,8 +133,13 @@ def as_calendar_objects(options = {})
     ScheduleRuleCalendarPresenter.new(self, options).to_json
   end
 
-  def discount_for(start_at, end_at)
-    percent_overlap(start_at, end_at) * discount_percent.to_f
+  def discount_for(start_at, end_at, price_group)
+    price_group_discount = PriceGroupDiscount.where(
+      price_group: price_group,
+      schedule_rule: self
+    ).first
```

Comment:
I think this logic could be more succint like:
```suggestion 
    price_group_discount = price_group_discounts.find_by(price_group: price_group)
```